clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1215,6 +1241,66 @@ class ArchiveFileHandler final : public FileHandler {
   Error WriteBundle(raw_fd_ostream &OS, MemoryBuffer &Input) override {
     llvm_unreachable("unsupported for the ArchiveFileHandler");
   }
+
+private:
```

Comment:
We are opening and parsing the file here an additional time. This can prove expensive. There must be a way to do this 'exclusion' on the fly using a single parse of the file. But we can do that as a subsequent work. Please add a TODO here.

Reply:
I thought about this optimizations.
There is a cost we should always consider is that this tool is from community. And we should resolve conflicts in sycl-web every time something comes from community in this functionality. So if we change the archive processing significantly in terms of a code then conflict resolving could turn into hard work for one who will maintain this stuff (it is me or somebody from our team).

Reply:
Here it is. It removes useless work for general case.

https://github.com/intel/llvm/pull/9572/commits/3707aa432f60f804cac765039c29f5fae60b43ed#diff-c29010597bfa221cf7e7cfc018e0b0a7bdae17bcf8d17b27be96f4cbd20ec28aR1293-R1295

Reply:
This is surely not a blocker for this PR. But we need to address this and atleast verify that this does not increase compile time by a lot. Thanks